test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -2083,12 +2083,17 @@ def decode_value(cls, value):
         of it, if it is made of alphanumeric values only, return
         it as a string, if it is of type size, i.e., with a memory
         unit such as b,kb,mb,gb then return the converted size to
-        kb without the unit
+        kb without the unit, if value is a dictionary string return
+        dictionary value
 
         :param value: attribute/resource value
         :type value: str or int
         :returns: int or float or string
         """
+        if re.search('(\{.*:.*[,]?.*\})', str(value)):
+            value = json.loads(eval(str(value)))
```

Comment:
unfortunately, it doesn't seem like this works either. It only works for dicts which have string keys, for dicts which have numeric keys, it doesn't work:
```

>>> mystr = '{1:"blah", 2:[1, "a", "blah"], 3: {1:2, 3:4}}'
>>> json.loads(mystr)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
>>> json.loads(eval(str(mystr)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not dict
```

How about using ast.literal_eval() instead?

Reply:
I am ok with `ast.literal_eval()`.